src/coreclr/System.Private.CoreLib/src/System/Reflection/TypeNameParser.CoreCLR.cs

```diff
@@ -173,13 +184,14 @@ private bool CheckTopLevelAssemblyQualifiedName()
             Justification = "TypeNameParser.GetType is marked as RequiresUnreferencedCode.")]
         [UnconditionalSuppressMessage("ReflectionAnalysis", "IL2075:UnrecognizedReflectionPattern",
             Justification = "TypeNameParser.GetType is marked as RequiresUnreferencedCode.")]
-        private Type? GetType(string typeName, ReadOnlySpan<string> nestedTypeNames, string? assemblyNameIfAny)
+        private Type? GetType(string escapedTypeName, // For nested types, it's Name. For other types it's FullName
+            ReadOnlySpan<string> nestedTypeNames, Metadata.AssemblyNameInfo? assemblyNameIfAny, string fullEscapedName)
```

Comment:
Are we creating the fullEscapedName just to pass it to the various rarely used paths? It is a de-optimization of the most common path.

The signature contract of the method that takes all names as unescaped, and creates the full name or escaped names as needed, is optimized for the most common path at the cost of making very rarely used paths more expensive.

Reply:
Great catch! I am passing `TypeName` now and `FullName` is accessed only when needed.